lib/miq_expression.rb

```diff
@@ -1404,24 +1398,26 @@ def to_arel(exp, tz)
     end
   end
 
-  def extract_where_values(klass, scope)
-    relation = ActiveRecord::Relation.new(klass, klass.arel_table, klass.predicate_builder)
-    relation = relation.instance_eval(&scope)
+  def subquery_for_contains(field, limiter_query)
+    return limiter_query if field.reflections.empty?
 
-    begin
-      # This is basically ActiveRecord::Relation#to_sql, only using our
-      # custom visitor instance
-
-      connection = klass.connection
-      visitor    = WhereExtractionVisitor.new(connection)
-
-      arel  = relation.arel
-      binds = relation.bound_attributes
-      binds = binds.collect(&:value_for_database)
-      binds.map! { |value| connection.quote(value) }
-      collect = visitor.accept(arel.ast, Arel::Collectors::Bind.new)
-      collect.substitute_binds(binds).join
-    end
+    # Remove the default scopes via `base_class`. The scope is already in the main query and not needed in the subquery
+    main_model = field.model.base_class
+    primary_attribute = main_model.arel_table[main_model.primary_key]
+
+    includes_associations = field.reflections.reverse.inject({}) { |i, k| {k.name => i} }
+    relation_query = main_model.select(primary_attribute)
+                               .joins(includes_associations)
+                               .where(limiter_query)
+
+    conn = main_model.connection
+    sql  = if ActiveRecord.version.to_s >= "5.2"
```

Comment:
You know what, I will concede on this for now since it is such a small area we are dealing with (instead of 50 lines of visitor code that was only for 5.1 compatibility).

That said, I think we should delete this right away as part of https://github.com/ManageIQ/manageiq/pull/20076 if we are planning to merge this into `master` right away.  I would prefer this conditional not sticking around forever.

Reply:
Yes, great idea.  Once this is merged, let's drop this conditional in #20076

Reply:
For this PR, is Travis set up to test BOTH Rails 5.1 and 5.2?  Or is the current PR being tested on Rails 5.1?  And #20076 will validate on Rails 5.2?

Reply:
This PR is still testing against Rails 5.1.

We have a few branches doing cross repo testing that have the branch from #20076 in it:

https://travis-ci.org/github/jrafanie/manageiq-cross_repo-tests/builds/670576839
https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/114

Which are validating it there, but we will confirm this works in 5.2 again when #20076 is rebased again, un-WIP'd, and ready for a final review.